src/raw/aws/fargate_inspector.go

```diff
@@ -153,3 +168,19 @@ func isECSARN(arn string) bool {
 
 	return strings.HasPrefix(arn, arnPrefix) && strings.Count(arn, ":") >= arnSections-1
 }
+
+// regionFromTask returns the aws region from the task ARN.
+// Example of task ARN: arn:aws:ecs:us-west-2:xxxxxxxx:task/ecs-local-cluster/37e873f6-37b4-42a7-af47-eac7275c6152
+func regionFromTask(taskARN string) string {
+	_, arnPrefix := resourceNameAndARNBase(taskARN)
+	return strings.Split(arnPrefix, ":")[3]
+}
+
+// resourceNameAndARNBase explodes a resource ARN and returns the resource's name and the base ARN prefix for the
+// account and region of the original resource.
+func resourceNameAndARNBase(resourceARN string) (resourceName string, arnPrefix string) {
+	arnPrefixAndType := resourceARN[:strings.Index(resourceARN, "/")-1]
```

Comment:
Can we use the strings.Split function instead? 
Only to avoid splitting and using indexes directly

Reply:
Both functions were taken (i.e. copy-pasted) from [nri-ecs](https://github.com/newrelic/nri-ecs/blob/main/internal/metadata/v3/v3.go), and were left untouched for consistency even when I do not particularly like the implementation of either. The one that uses split, for example, does not perform any kind of bound-checking and relies on the passed string to be an arn.

If we are ok with changing them, I'd vote for rewriting both of them to use `strings.Split` and proper bound checking. What do you think?